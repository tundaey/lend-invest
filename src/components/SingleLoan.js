import React from 'react';
import {Badge} from 'react-bootstrap'

const SingleLoan = (props) => (
    <tr className={props.loan.invested? 'invested-row ': ''}
        onClick={(event)=> props.onClick(event, props.loan)}>
        <td>{props.loan.id}</td>
        <td>{props.loan.title}</td>
        <td>{props.loan.tranche}</td>
        <td>{props.loan.available}</td>
        <td>{props.loan.annualised_return}</td>
        <td>{props.loan.term_remaining}</td>
        <td>{props.loan.ltv}</td>
        <td>{props.loan.amount}</td>
        <td><Badge bsClass={props.loan.invested? 'invested-badge': 'badge'}>{props.loan.invested? 'Invested': 'Eligible'}</Badge></td>
    </tr>
)

export default SingleLoan