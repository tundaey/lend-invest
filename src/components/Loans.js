import React from 'react';
import { Table} from 'react-bootstrap'
import SingleLoan from './SingleLoan'

const Loans = (props)=> (
    <div>
        <Table bordered condensed hover responsive>
            <thead>
                <tr>
                    <th>#</th>
                    <th>Title</th>
                    <th>Tranche</th>
                    <th>Available</th>
                    <th>Annualised Return</th>
                    <th>Term remaining</th>
                    <th>Ltv</th>
                    <th>Amount</th>
                </tr>
            </thead>
            <tbody>
                {props.loans.map((loan)=> <SingleLoan onClick={props.openModal} key={loan.id} loan={loan}/>)}
            </tbody>
        </Table>
    </div>
)

export default Loans