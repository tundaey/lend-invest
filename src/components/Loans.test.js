import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import {loans} from '../current-loans.json'
import Loans from './Loans'
import SingleLoan from './SingleLoan'

test('User should be able to see all current loans', ()=> {
    const wrapper = shallow(<Loans loans={loans}/>)
    const element = wrapper.find(SingleLoan)
    const firstLoanId = loans[0].id
    expect(element.length).toBe(3)
    expect(element.get(0).props.loan.id).toBe(firstLoanId)
})

test('Loans component renders a snapshot properly', ()=>{
    const tree = renderer.create(<Loans loans={loans}/>).toJSON();
    expect(tree).toMatchSnapshot(); 
})