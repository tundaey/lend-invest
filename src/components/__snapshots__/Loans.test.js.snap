// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loans component renders a snapshot properly 1`] = `
<div>
  <div
    className="table-responsive"
  >
    <table
      className="table table-bordered table-condensed table-hover"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Title
          </th>
          <th>
            Tranche
          </th>
          <th>
            Available
          </th>
          <th>
            Annualised Return
          </th>
          <th>
            Term remaining
          </th>
          <th>
            Ltv
          </th>
          <th>
            Amount
          </th>
          <th>
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          onClick={[Function]}
        >
          <td>
            1
          </td>
          <td>
            Voluptate et sed tempora qui quisquam.
          </td>
          <td>
            A
          </td>
          <td>
            11,959
          </td>
          <td>
            8.60
          </td>
          <td>
            864000
          </td>
          <td>
            48.80
          </td>
          <td>
            85,754
          </td>
          <td>
            <span
              className="badge"
            >
              Eligible
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
        >
          <td>
            5
          </td>
          <td>
            Consectetur ipsam qui magnam minus dolore ut fugit.
          </td>
          <td>
            B
          </td>
          <td>
            31,405
          </td>
          <td>
            7.10
          </td>
          <td>
            1620000
          </td>
          <td>
            48.80
          </td>
          <td>
            85,754
          </td>
          <td>
            <span
              className="badge"
            >
              Eligible
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
        >
          <td>
            12
          </td>
          <td>
            Dolores repudiandae ut voluptas unde laborum quaerat et sapiente.
          </td>
          <td>
            C
          </td>
          <td>
            12,359
          </td>
          <td>
            4.80
          </td>
          <td>
            879000
          </td>
          <td>
            48.80
          </td>
          <td>
            85,754
          </td>
          <td>
            <span
              className="badge"
            >
              Eligible
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
